### ISSUE TITLE
VPN example broken on master

### ISSUE THREAD

**Issue by nickolaev** [networkservicemesh/networkservicemesh#733]
We have the VPN example broken in master. This is how to reproduce:

```
make vagrant-start
make k8s-infra-deploy
make k8-vpn-deploy
make k8s-check
```

You will see that the connections are not stable, created and then disappear. Not sure what the exact problem is but reverting 0254f6e seems to help.

**Comment by nickolaev** [comment]
#732 adds VPN integration tests for the CI to demonstrate the problem.

**Comment by haiodo** [comment]
Hi Will check today, do not want to rollback changes, since have more coming after them with healing. Probable issue has same source as I fight right now, with dataplane crashes.

**Comment by nickolaev** [comment]
Thanks, @haiodo. Of course, rollback is an extreme measure and will try to avoid it. I hope you'll be able to find something.

**Comment by haiodo** [comment]
Dataplane is responding with: 

```
time="2019-02-09T11:05:54Z" level=error msg="Dataplane request failed: rpc error: code = Unknown desc = error registering socket file name /var/lib/networkservicemesh/nsmwA0Tt3B0O/memif.sock (ID 2): VPPApiError: Invalid argument (-73)"
time="2019-02-09T11:05:54Z" level=info msg="ClientConnectionDeleted: &{1 <nil> <nil> network_service:<name:\"secure-intranet-connectivity\" payload:\"IP\" > network_service_manager:<name:\"kube-worker\" url:\"10.44.0.4:5001\" last_seen:<seconds:1549710149 nanos:910059245 > state:\"RUNNING\" > networkservice_endpoint:<network_service_name:\"secure-intranet-connectivity\" network_service_manager_name:\"kube-worker\" endpoint_name:\"secure-intranet-connectivitymgtdv\" labels:<key:\"app\" value:\"firewall\" > labels:<key:\"networkservicename\" value:\"secure-intranet-connectivity\" > state:\"RUNNING\" >  0xc000084d20 false}"
time="2019-02-09T11:05:54Z" level=error msg="rpc error: code = Unknown desc = error registering socket file name /var/lib/networkservicemesh/nsmwA0Tt3B0O/memif.sock (ID 2): VPPApiError: Invalid argument (-73)"
```

**Comment by haiodo** [comment]
VPPApiError: Invalid argument (-73)"

Have no idea, what is going wrong, will check commit changes.

**Comment by haiodo** [comment]
It seems finally find issue,

Dataplane configuration for memif is missing workspace passed to it.

with applied 
https://github.com/networkservicemesh/networkservicemesh/commit/0254f6e159d5504a39db725f5839d084ae7109c3

```
level=info msg="Sending DataChange to vppagent: Interfaces:<name:\"SRC-1\" type:VXLAN_TUNNEL enabled:true vxlan:<src_address:\"10.44.0.3\" dst_address:\"10.32.0.7\" vni:2 > > Interfaces:<name:\"DST-1\" type:MEMORY_INTERFACE enabled:true memif:
<socket_filename:\"/var/lib/networkservicemesh/nsmwA0Tt3B0O/memif.sock\" >
 > XCons:<receive_interface:\"SRC-1\" transmit_interface:\"DST-1\" > XCons:<receive_interface:\"DST-1\" transmit_interface:\"SRC-1\" > "
```

Without applied 
https://github.com/networkservicemesh/networkservicemesh/commit/0254f6e159d5504a39db725f5839d084ae7109c3

```

level=info msg="Sending DataChange to vppagent: Interfaces:<name:\"SRC-1\" type:VXLAN_TUNNEL enabled:true vxlan:<src_address:\"10.44.0.3\" dst_address:\"10.32.0.6\" vni:2 > > Interfaces:<name:\"DST-1\" type:MEMORY_INTERFACE enabled:true memif:
<socket_filename:\"/var/lib/networkservicemesh/nsm-2/nsmKti2LGKuY/memif.sock\" >
 > XCons:<receive_interface:\"SRC-1\" transmit_interface:\"DST-1\" > XCons:<receive_interface:\"DST-1\" transmit_interface:\"SRC-1\" > "
```

Will prepare MR to fix issue in master in few hours.

**Comment by nickolaev** [comment]
@haiodo that's a great finding. Actually, #732 is indirectly testing the MEMIF since the firewall uses it. Looking forward to get your patch.

**Comment by nickolaev** [comment]
@haiodo is this something that can break your pathces:

https://github.com/networkservicemesh/networkservicemesh/blob/master/sdk/endpoint/composite/client.go#L58

It is not part of your patch but since it is labeled as "hack" it may be a problem. What do you think?

**Comment by nickolaev** [comment]
Closing as fixed by #734

**Comment by haiodo** [comment]
NSMD pass a proper workspace by himself, so this code you mentioned I think do nothing finally:-)
